### PR TITLE
SVProgressHUD - activity indicator

### PR DESCRIPTION
**Pull request by kobsonaut**

### CODE REVIEW

File: TOZ_iOS/GalleryViewController.swift [hunk 1]
```diff
@@ -64,6 +66,14 @@ class GalleryViewController: UIViewController, UITableViewDataSource, UITableVie
                 print ("\(error)")
             }
         }
-            listOfAnimalsOperation.start()
+        listOfAnimalsOperation.start()
+    }
+    func runActivityIndicator() {
```
Comment (filip-zielinski) [inline TOZ_iOS/GalleryViewController.swift]: I think `SVProgressHUD.show()` should be executed before a time consuming task starts and `SVProgressHUD.dismiss()` should be executed when it's done. Right now it looks like progress indicator is shown and **immediately** dismissed so it really can't be seen. And it is called after `getListOfAnimals()`.

File: TOZ_iOS/HelpFinanceViewController.swift [hunk 2]
```diff
@@ -50,4 +51,12 @@ class HelpFinanceViewController: UIViewController {
         organizationInfoOperation.start()
     }
 
+    func runActivityIndicator() {
```
Comment (filip-zielinski) [inline TOZ_iOS/HelpFinanceViewController.swift]: Same comment as in GalleryViewController.

File: TOZ_iOS/HelpViewController.swift [hunk 3]
```diff
@@ -17,5 +18,15 @@ class HelpViewController: UIViewController {
         helpScrollView.backgroundColor = Color.Background.primary
         helpVolunteerView.backgroundColor = Color.HelpUIViews.background
         helpFinanceView.backgroundColor = Color.HelpUIViews.background
+        runActivityIndicator()
+    }
+
+    func runActivityIndicator() {
```
Comment (filip-zielinski) [inline TOZ_iOS/HelpViewController.swift]: Same comment as in GalleryViewController.

File: TOZ_iOS/GalleryViewController.swift (hunk 1 above)
Comment (kobsonaut) [inline TOZ_iOS/GalleryViewController.swift]: Yes. You're right, SVProgressHUD should be shown before a time consuming task and dismissed right after, but in our situation it's page loading. If you want to see how it works just mark `SVProgressHUD.dismiss() ` as a comment or add a some delay as I did.

File: TOZ_iOS/GalleryViewController.swift [hunk 4]
```diff
@@ -64,6 +66,14 @@ class GalleryViewController: UIViewController, UITableViewDataSource, UITableVie
...
+        SVProgressHUD.show()
+        DispatchQueue.global().async {
+            DispatchQueue.main.async {
+                SVProgressHUD.dismiss(withDelay: 0.5)
+            }
```
Comment (piotr-tobolski) [inline TOZ_iOS/GalleryViewController.swift]: This doesn't make any sense `SVProgressHUD.show()` should be just before `listOfAnimalsOperation.start()` and `SVProgressHUD.dismiss()` should be inside `resultCompletion` block

File: TOZ_iOS/SignUpViewController.swift [hunk 5]
```diff
@@ -58,23 +59,28 @@ class SignUpViewController: UIViewController {
...
+                            SVProgressHUD.dismiss()
                             let alert = UIAlertController(title: "Sukces", message: "Zgłoszenie zostało wysłane", preferredStyle: UIAlertControllerStyle.alert)
                             alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: nil))
                             self.present(alert, animated: true, completion: nil)
                         }
                     }
                 }
             }
+        } else {
+            SVProgressHUD.dismiss()
```
Comment (piotr-tobolski) [inline TOZ_iOS/SignUpViewController.swift]: Validating text field doesn't take any time so showing HUD should be just before `signUpOperation.start`. This way you won't have to dismiss it almost at the same time it was shown

File: TOZ_iOS/SignUpViewController.swift [hunk 6]
```diff
@@ -58,23 +59,28 @@ class SignUpViewController: UIViewController {
...
             if let signUpOperation = self.signUpOperation {
                 signUpOperation.start { succes in
                     DispatchQueue.main.async {
                         if succes == false {
+                            SVProgressHUD.dismiss()
                             let alert = UIAlertController(title: "Ups", message: "Wystąpił błąd, spróbuj ponownie później.", preferredStyle: UIAlertControllerStyle.alert)
                             alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: nil))
                             self.present(alert, animated: true, completion: nil)
                         } else {
+                            SVProgressHUD.dismiss()
```
Comment (piotr-tobolski) [inline TOZ_iOS/SignUpViewController.swift]: just call it once before the `if`

File: TOZ_iOS/NewsViewController.swift [hunk 7]
```diff
@@ -49,19 +50,21 @@ class NewsViewController: UIViewController, UITableViewDataSource, UITableViewDe
...
             switch result {
             case .success(let newsList):
                 DispatchQueue.main.async {
                     self.localNewsList = newsList
                     self.newsTableView.reloadData()
+                    SVProgressHUD.dismiss()
                 }
             case .failure(let error):
                 print ("\(error)")
+                SVProgressHUD.dismiss()
```
Comment (piotr-tobolski) [inline TOZ_iOS/NewsViewController.swift]: just call it once before the `switch`

File: TOZ_iOS/LoginViewController.swift [hunk 8]
```diff
@@ -20,18 +21,21 @@ class LoginViewController: UIViewController, UITextFieldDelegate {
...
                 DispatchQueue.main.async {
+                    SVProgressHUD.dismiss()
                     BackendAuth.shared.setToken(successfullSignIn.jwt)
                     if let token = BackendAuth.shared.token {
                         print("Token >\(token)< successfully set for email \(self.emailInput.text)")
                     }
                 }
             case .failure(let error):
                 DispatchQueue.main.sync {
+                    SVProgressHUD.dismiss()
```
Comment (piotr-tobolski) [inline TOZ_iOS/LoginViewController.swift]: just call it once before the `switch`

File: TOZ_iOS/LoginViewController.swift [hunk 9]
```diff
@@ -41,6 +45,8 @@ class LoginViewController: UIViewController, UITextFieldDelegate {
         }
         if self.emailInput.isValid && self.passwordInput.isValid {
             signInOperation?.start()
+        } else {
+            SVProgressHUD.dismiss()
```
Comment (piotr-tobolski) [inline TOZ_iOS/LoginViewController.swift]: Validating text field doesn't take any time so showing HUD should be just before `signInOperation.start()`. This way you won't have to dismiss it almost at the same time it was shown

File: TOZ_iOS/HelpViewController.swift [hunk 10]
```diff
@@ -6,6 +6,7 @@
 //
 
 import UIKit
+import SVProgressHUD
```
Comment (piotr-tobolski) [inline TOZ_iOS/HelpViewController.swift]: ?

File: TOZ_iOS/HelpFinanceViewController.swift [hunk 11]
```diff
@@ -42,12 +40,13 @@ class HelpFinanceViewController: UIViewController {
                     self.financeDescription.text = "Można pomóc wpłacając darowiznę na konto Szczecińskiej Fundacji Pomocy Zwierzętom Wszelakim albo przekazując dary rzeczowe. Szczegóły poniżej:"
                     self.financeInfo.text = localOrganization.name + "\n" + postCode + " " + city + "\n" + street + " " + houseNumber + apartmentNumber
                     self.accountNumberAndName.text = localOrganization.accountNumber + "\n" + bankName
+                    SVProgressHUD.dismiss()
                 }
             case .failure(let error):
                 print ("\(error)")
+                SVProgressHUD.dismiss()
```
Comment (piotr-tobolski) [inline TOZ_iOS/HelpFinanceViewController.swift]: just call it once before the `switch`

File: TOZ_iOS/GalleryViewController.swift [hunk 12]
```diff
@@ -54,16 +55,19 @@ class GalleryViewController: UIViewController, UITableViewDataSource, UITableVie
...
         listOfAnimalsOperation.resultCompletion = { result in
             switch result {
             case .success(let listOfAnimals):
                 DispatchQueue.main.async {
                     self.animalsArray = listOfAnimals
+                    SVProgressHUD.dismiss()
                 }
             case .failure(let error):
                 print ("\(error)")
+                SVProgressHUD.dismiss()
```
Comment (piotr-tobolski) [inline TOZ_iOS/GalleryViewController.swift]: just call it once before the `switch`

Comment (piotr-tobolski) [inline TOZ_iOS/GalleryViewController.swift]: you forgot about this one

**Pull request merged**